cmd/goal/node_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var isNum = regexp.MustCompile(`^[0-9]+$`)
+var isAlnum = regexp.MustCompile(`^[a-zA-Z0-9_]*$`)
+
+func TestGetMissingCatchpointLabel(t *testing.T) {
+	tests := []struct {
+		name        string
+		URL         string
+		expectedErr string
+		statusCode  int
+	}{
+		{
+			"bad request",
+			"",
+			"400 Bad Request",
+			http.StatusBadRequest,
+		},
+		{
+			"forbidden request",
+			"",
+			"403 Forbidden",
+			http.StatusForbidden,
+		},
+		{
+			"page not found",
+			"",
+			"404 Not Found",
+			http.StatusNotFound,
+		},
+		{
+			"bad gateway",
+			"",
+			"502 Bad Gateway",
+			http.StatusBadGateway,
+		},
+		{
+			"mainnet catchpoint",
+			"https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/mainnet/latest.catchpoint",
+			"",
+			http.StatusAccepted,
+		},
+		{
+			"betanet catchpoint",
+			"https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/betanet/latest.catchpoint",
+			"",
+			http.StatusAccepted,
+		},
+		{
+			"testnet catchpoint",
+			"https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/testnet/latest.catchpoint",
+			"",
+			http.StatusAccepted,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				http.Error(w, test.expectedErr, test.statusCode)
+			}))
+			defer ts.Close()
+
+			if test.expectedErr != "" {
+				test.URL = ts.URL
+			}
+
+			label, err := getMissingCatchpointLabel(test.URL)
+
+			if test.expectedErr != "" {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.expectedErr)
+			} else {
+				assert.True(t, isNum.MatchString(label[:8]))
+				assert.True(t, isAlnum.MatchString(label[9:]))
```

Comment:
This test would break when we get another digit in the round number, need to look for the index of #
```suggestion
                                // check for properly formatted catchpoint label.
                                separatorIndex := strings.Index(label, "#")
                                assert.NotEqual(t, 0, separatorIndex)
                                expectedLength := separatorIndex + len(types.ZeroAddress.String()) + 1
                                assert.Equal(t, expectedLength, len(label))
				assert.True(t, isNum.MatchString(label[:separatorIndex]))
				assert.True(t, isAlnum.MatchString(label[separatorIndex+1:]))
```